### PR TITLE
openblas: add patches to build with clang

### PR DESCRIPTION
**Pull request by boomanaiden154**
Certain builds/installs of clang on some systems are running into these issues. The libomp issue is being tracked upstream and a patch has been opened upstream in OpenBLAS in regards to the make variable overwrite.

### CODE REVIEW

**Comment by boomanaiden154** [comment]
Upstreaming one of the patches in https://github.com/xianyi/OpenBLAS/pull/4171. The other is due to an upstream OpenMP runtime issue that is being tracked at https://github.com/llvm/llvm-project/issues/63908.

File: var/spack/repos/builtin/packages/openblas/package.py [hunk 1]
```diff
@@ -184,6 +184,17 @@ class Openblas(MakefilePackage):
         when="@0.3.21 %gcc@:9",
     )
 
+    # Some installations of clang and libomp have non-standard locations for
+    # libomp. OpenBLAS adds the correct linker flags but overwrites the
+    # variables in a couple places, causing link-time failures.
+    patch("openblas_append_lflags.patch", when="threads=openmp %clang")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/openblas/package.py]: You have to restrict to the versions where it's necessary.

https://github.com/OpenMathLib/OpenBLAS/commit/9a0e9c8b69ca1441ac7949a73de237cbff95812e

From where does it apply? Looks like it's in  v0.3.24. Probably fine to apply the patch for all compilers.

**Review by haampie — CHANGES_REQUESTED** [review]
Patch needs bounds

File: var/spack/repos/builtin/packages/openblas/package.py (hunk 1 above)
Comment (boomanaiden154) [reply var/spack/repos/builtin/packages/openblas/package.py]: Yep. Meant to update it when the patch actually landed upstream but didn't get around to it. Fixed now.

**Review by becker33 — APPROVED** [review]

**Comment by becker33** [comment]
@haampie can you confirm the changes satisfied your review?

**Review by haampie — APPROVED** [review]

**Pull request merged**